src/arrayMethodJoin.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const mySeparator = separator === undefined ? ',' : separator;
```

Comment:
let's assign default separator while describing function arguments instead of creating separate check